### PR TITLE
refactor: extract env file handling

### PR DESCRIPTION
**Pull request by augustoccesar**
### Description
Move env file handling to own module so that we can better test its behaviour and have it contained.

### Note
We need the service name on the env file handling just so that we can use that while building the `CliError`. It feels that maybe we can return another type of error by the env handlers and leave the translation into a `CliError` as a responsibility of the caller. But maybe we can do that on another PR.

### CODE REVIEW

File: linkup-cli/src/env_files.rs [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+                format!("Failed to write .env file: {}", e),
+            )
+        })?;
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
```
Comment (ostenbom) [inline linkup-cli/src/env_files.rs:111]: Yeeesss this is really cool. Easier to write tests the way you managed to pull it out. Nice one!!

**Review by ostenbom — COMMENTED** [review]
Super nice to have that pulled out and tested - the logic there had some hairy edge cases for sure!

**Review by ostenbom — APPROVED** [review]
